### PR TITLE
Yr/azure sentinel/add 2 new commands/ciac 6573

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6573

## Description
Added 2 new commands: 
*azure-sentinel-subscriptions-list
*azure-sentinel-resource-group-list

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 1]
```diff
@@ -148,7 +148,8 @@ def http_request(self, method, url_suffix=None, full_url=None, params=None, data
         """
         if not full_url:
             params = params or {}
-            params['api-version'] = API_VERSION
+            if not params.get('api-version'):
```
Comment (ShacharKidor) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:151]: Why do we need this condition? do we expect API version param?

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 2]
```diff
@@ -1795,6 +1796,66 @@ def delete_alert_rule_command(client: AzureSentinelClient, args: Dict[str, Any])
...
+            tag = json.loads(tag)
+            tag_name = next(iter(tag))
+            tag_value = tag[tag_name]
+            filter_by_tag = f"tagName eq '{tag_name}' and tagValue eq '{tag_value}'"
+        except Exception as e:
+            raise ValueError(('Invalid tag format, please use the following format: `{"key_name":"value_name"}`', e))
+
+    data_from_response = []
+    for subscription_id in subscription_id_list:
+        full_url = f'https://management.azure.com/subscriptions/{subscription_id}/resourcegroups?$filter={filter_by_tag}&$top={limit}&api-version=2021-04-01'   # noqa: E501
```
Comment (ShacharKidor) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:1840]: why do we use a different api version in each command?

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 3]
```diff
@@ -1795,6 +1796,66 @@ def delete_alert_rule_command(client: AzureSentinelClient, args: Dict[str, Any])
...
 
 
+def list_subscriptions_command(client: AzureSentinelClient) -> CommandResults:
+
+    full_url = 'https://management.azure.com/subscriptions?api-version=2020-01-01'
+
+    response = client.http_request('GET', full_url=full_url)
+    data_from_response = response.get('value', [])
+
+    readable_output = tableToMarkdown(
```
Comment (ShacharKidor) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:1806]: if the list is empty - consider changing the RH to - no subscriptions was found - or something like that.

**Review by ShacharKidor — CHANGES_REQUESTED** [review]
Great Work!

Please see my comments below. 
Also, did you add the new commands to the TPB? if yes - please add a "great run" screenshot to the PR's body.

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 1 above)
Comment (RosenbergYehuda) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:151]: same as in key vault, this is the http_request function params, not the demisto.params()

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 2 above)
Comment (RosenbergYehuda) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:1840]: by design

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 1 above)
Comment (ShacharKidor) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:151]: I can see that the API version of the new lost commands is passed as part of the full URL. In that case why do we need this condition? 
`if not params.get('api-version'):`

**Review by ShacharKidor — CHANGES_REQUESTED** [review]
Good work! 

Comments regarding the API module changes were added in this PR - [pull/26943](https://github.com/demisto/content/pull/26943).

Please add a screen shot of a green run of the edited TPB. 

Please check the API version issue we talked about (explained in the relevant comment).

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 1 above)
Comment (RosenbergYehuda) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:151]: Fixed.

**Review by kobymeir — APPROVED** [review]

**Review by ShacharKidor — APPROVED** [review]

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 4]
```diff
@@ -3025,6 +3025,65 @@ script:
...
+      description: The name of the resource.
+      type: String
+    - contextPath: AzureSentinel.Subscription.authorizationSource
+      description: The authorization source of the resource.
+      type: String
+    - contextPath: AzureSentinel.Subscription.managedByTenants
+      description: The managedByTenants of the resource.
+      type: String
+    - contextPath: AzureSentinel.Subscription.tenetId
+      description: The tenetId of the resource.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:3044]:
```suggestion
      description: The tenet ID of the resource.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 5]
```diff
@@ -3025,6 +3025,65 @@ script:
...
+      description: Fully qualified resource ID for the resource.
+      type: String
+    - contextPath: AzureSentinel.Subscription.displayName
+      description: The name of the resource.
+      type: String
+    - contextPath: AzureSentinel.Subscription.authorizationSource
+      description: The authorization source of the resource.
+      type: String
+    - contextPath: AzureSentinel.Subscription.managedByTenants
+      description: The managedByTenants of the resource.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:3041]:
```suggestion
      description: The subscriptions that are managed by tenants of the resource.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 6]
```diff
@@ -3025,6 +3025,65 @@ script:
...
+      type: String
+    - contextPath: AzureSentinel.Subscription.tenetId
+      description: The tenetId of the resource.
+      type: String
+    - contextPath: AzureSentinel.Subscription.state
+      description: The state of the resource.
+      type: String
+    - contextPath: AzureSentinel.Subscription.subscriptionPolicies
+      description: The subscriptionPolicies of the resource.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:3050]:
```suggestion
      description: The subscription policies of the resource.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 7]
```diff
@@ -3025,6 +3025,65 @@ script:
...
+    - name: subscription_id
+      description: "The subscription ID. Note: The integration default Subscription ID will be used unless this argument is provided."
+      type: String
+      required: false
+    - name: tag
+      description: A single tag in the form of '{"Tag Name":"Tag Value"}' to filter the list by.
+      isArray: false
+      type: String
+    - name: limit
+      description: Limit on the number of resource groups to return.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:3064]:
```suggestion
      description: The maximum number of resource groups to return.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/README.md [hunk 8]
```diff
@@ -2397,3 +2397,137 @@ Updates an alert rule.
...
+There are no input arguments for this command.
+
+#### Context Output
+
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| AzureSentinel.SubscriptionId | String | Fully qualified resource ID for the resource. | 
+| AzureSentinel.Subscription.displayName | String | The name of the resource. | 
+| AzureSentinel.Subscription.authorizationSource | String | The authorization source of the resource. | 
+| AzureSentinel.Subscription.managedByTenants | String | The managedByTenants of the resource. |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/README.md:2421]:
```suggestion
| AzureSentinel.Subscription.managedByTenants | String | The subscriptions that are managed by tenants of the resource. | 
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/README.md [hunk 9]
```diff
@@ -2397,3 +2397,137 @@ Updates an alert rule.
...
+
+#### Context Output
+
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| AzureSentinel.SubscriptionId | String | Fully qualified resource ID for the resource. | 
+| AzureSentinel.Subscription.displayName | String | The name of the resource. | 
+| AzureSentinel.Subscription.authorizationSource | String | The authorization source of the resource. | 
+| AzureSentinel.Subscription.managedByTenants | String | The managedByTenants of the resource. | 
+| AzureSentinel.Subscription.tenetId | String | The tenetId of the resource. |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/README.md:2422]:
```suggestion
| AzureSentinel.Subscription.tenetId | String | The tenet ID of the resource. | 
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/README.md [hunk 10]
```diff
@@ -2397,3 +2397,137 @@ Updates an alert rule.
...
+
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| AzureSentinel.SubscriptionId | String | Fully qualified resource ID for the resource. | 
+| AzureSentinel.Subscription.displayName | String | The name of the resource. | 
+| AzureSentinel.Subscription.authorizationSource | String | The authorization source of the resource. | 
+| AzureSentinel.Subscription.managedByTenants | String | The managedByTenants of the resource. | 
+| AzureSentinel.Subscription.tenetId | String | The tenetId of the resource. | 
+| AzureSentinel.Subscription.state | String | The state of the resource. | 
+| AzureSentinel.Subscription.subscriptionPolicies | String | The subscriptionPolicies of the resource. |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/README.md:2424]:
```suggestion
| AzureSentinel.Subscription.subscriptionPolicies | String | The subscription policies of the resource. | 
```

**Comment by ShirleyDenkberg** [comment]
@kobymeir @ShacharKidor @michal-dagan Doc review completed.

**Pull request merged**